astropy/coordinates/tests/test_frames.py

```diff
@@ -259,7 +260,11 @@ def test_no_data_nonscalar_frames():
             obstime=Time("2012-01-01") + np.arange(10.0) * u.day,
             temperature=np.ones((3,)) * u.deg_C,
         )
-    assert "inconsistent shapes" in str(exc.value)
```

Comment:
Actually, looking at this again, a question: I assume the "inconsistent shapes" appear in the text the user sees, since otherwise it is not very useful. If so, why isn't it in `str(exc)`? Would it be in `repr(exc)`? Ideally, we test as closely as possible what the user sees, not how it is implemented that they get to see it.

Reply:
I learned this while updating this test but notes are apparently not included in `__str__` or `__repr__`, but this is an implementation detail not visible by a user when an exception is raised.

Reply:
Ah, so the user sees it, but `str` or `repr` doesn't show it. That seems like a bug in python to me, or arguably `pytest`, which should really just match what the user sees. Anyway, for now this will have to do!

Reply:
I was curious so I looked it up and I'm not convinced it can be called a bug given [that `str()` is documented to represent instance arguments](https://docs.python.org/3/library/exceptions.html#BaseException). A shame that it forces us to test implantation details but I guess that's the price to pay for backward compatibility.

Reply:
Yes, I was just looking up things myself too, and came to the same conclusion. So, not a python bug, nor a pytest bug if one takes the description of `match` literally, though I think it might sense for `pytest` to support matching on anything the user would see. 
...
But then, of course I found that others had thought the same way, so newer pytest already includes `__notes__` in what it matches: https://github.com/pytest-dev/pytest/pull/11227. I think this has entered pytest 8.0 (the first release after the merge was 7.4.1, but seems unlikely it was included in a bugfix release).

In the meantime, maybe add a comment and link this PR in #15180?

Reply:
Excellent, thank you !